frontend/services/sources.go

```diff
@@ -319,29 +320,53 @@ func updateAnnotations(annotations map[string]string, reportedName string) map[s
 	return annotations
 }
 
-func GetSourceCRD(ctx context.Context, nsName string, workloadName string, workloadKind WorkloadKind) (*v1alpha1.Source, error) {
-	if workloadKind != WorkloadKindDeployment && workloadKind != WorkloadKindStatefulSet && workloadKind != WorkloadKindDaemonSet {
-		return nil, errors.New("unsupported workload kind " + string(workloadKind))
+func GetSourceCRDs(ctx context.Context, args ...interface{}) ([]*v1alpha1.Source, error) {
+	var nsName, workloadName string
+	var workloadKind WorkloadKind
+	if len(args) > 0 {
+		nsName, _ = args[0].(string)
+	}
+	if len(args) > 1 {
+		workloadName, _ = args[1].(string)
+	}
+	if len(args) > 2 {
+		workloadKind, _ = args[2].(WorkloadKind)
+		if workloadKind != WorkloadKindDeployment && workloadKind != WorkloadKindStatefulSet && workloadKind != WorkloadKindDaemonSet {
+			return nil, errors.New("unsupported workload kind " + string(workloadKind))
+		}
 	}
 
-	selector := labels.SelectorFromSet(labels.Set{
-		"odigos.io/workload-name":      workloadName,
-		"odigos.io/workload-namespace": nsName,
-		"odigos.io/workload-kind":      string(workloadKind),
-	})
+	labelsSet := labels.Set{}
+	if nsName != "" {
+		labelsSet["odigos.io/workload-namespace"] = nsName
+	}
+	if workloadName != "" {
+		labelsSet["odigos.io/workload-name"] = workloadName
+	}
+	if string(workloadKind) != "" {
+		labelsSet["odigos.io/workload-kind"] = string(workloadKind)
+	}
```

Comment:
what happen if one of the vars is empty?

is'nt it the same to do it like this?
```suggestion
	    labelsSet := labels.Set{
        "odigos.io/workload-namespace": nsName,
        "odigos.io/workload-name":      workloadName,
        "odigos.io/workload-kind":      string(workloadKind),
    }

    sourceList, err := kube.DefaultClient.OdigosClient.Sources(consts.DefaultOdigosNamespace).List(ctx, metav1.ListOptions{
        LabelSelector: labels.SelectorFromSet(labelsSet).String(),
    })
    if err != nil {
        return fmt.Errorf("failed to list sources: %w", err)
    }
```

Reply:
the args are optional, for cases when you need more than 1 source (like the whole list)